drivers/gpio/gpio_utils.h

```diff
@@ -11,6 +11,9 @@
 #ifndef ZEPHYR_DRIVERS_GPIO_GPIO_UTILS_H_
 #define ZEPHYR_DRIVERS_GPIO_GPIO_UTILS_H_
 
+#define GPIO_AVAILABLE_PINS(ngpios) ((ngpios == 32U)		\
+				     ? ((gpio_port_pins_t)(-1))	\
+				     : (BIT(ngpios) - 1U))
```

Comment:
No, it has to be this way.  `BIT(32)` produces an overflow when `1UL << 32` is evaluated and unsigned long is a 32-bit value.

Reply:
I misread the purpose of the code. But in this case it would be cleaner to cast to long long or uintmax_t. This will be computed during the compilation time, shouldn't influence the code size.